docs/pages/machine-id/troubleshooting.mdx

```diff
@@ -0,0 +1,286 @@
+---
+title: Machine ID Troubleshooting Guide
+description: Troubleshooting common issues with Machine ID
+---
+
+## A bot failed to renew a certificate due to a "generation mismatch", what does this mean?
+
+### Symptoms
+
+The bot will log an error like this:
+
+```text
+ERROR: renewable cert generation mismatch: stored=3, presented=2
+```
+
+Subsequent connection attempts by the bot may see errors like the following:
+```text
+ERROR: failed direct dial to auth server: auth API: access denied [00]
+"\tauth API: access denied [00], failed dial to auth server through reverse tunnel: Get \"https://teleport.cluster.local/v2/configuration/name\": Get \"https://example.com:3025/webapi/find\": x509: cannot validate certificate for example.com because it doesn't contain any IP SANs"
+"\tGet \"https://teleport.cluster.local/v2/configuration/name\": Get \"https://example.com:3025/webapi/find\": x509: cannot validate certificate for example.com because it doesn't contain any IP SANs"
+```
+
+In particular, note the message `auth API: access denied`. The Teleport auth
+server will provide some additional context:
+```text
+[AUTH]      WARN lock targeting User:"bot-example" is in force: The bot user "bot-example" has been locked due to a certificate generation mismatch, possibly indicating a stolen certificate. auth/apiserver.go:224
+```
+
+### Explanation
+
+Machine ID (with token-based joining) uses a certificate generation counter to
+detect potentially stolen renewable certificates. Each time a bot fetches a new
+renewable certificate, the counter is incremented and stored on the Teleport
```

Comment:
> the counter is incremented and stored on the Teleport
Auth Server and embedded in the certificate itself.

Suggesting something like this to tighten this up and clarify the architecture:

```markdown
the Auth Service increments the counter, stores it on the
backend, and embeds the counter in the certificate.
```

Reply:
That's much better, thanks! I think I'll add a small extra detail to mention that we embed "[a copy] of the counter in the certificate" to imply the cert isn't the source of truth (one of these days I need to write up an architecture doc).